### DISCUSSION TITLE
Row create partial default

### DISCUSSION THREAD

**Discussion by sdriffill-tm** [KevinVandy/material-react-table#1165] [Ideas]
## Background
I'm using Material React Table to create an editable grid with creating row functionality. I want to populate only a subset of the columns on my new blanks rows with a default value, in my case a column called `online` which holds a `boolean`, like so:
```tsx
renderTopToolbarCustomActions={({ table }) => (
  <Button
    onClick={() =>
      table.setCreatingRow(
        createRow(table, { online: true }),
      )
    }
  >
    + Add Row
  </Button>
)}
```

## Problem
I'm getting a type error on this line: `createRow(table, { online: true })`.
```ts
Argument of type '{ online: true; }' is not assignable to parameter of type <my specific MRT_RowData type>.
Type '{ online: true; }' is missing the following properties from type <my specific MRT_RowData type>: <the other properties>.ts(2345)
```

Whilst the second param to `createRow` is optional, if provided it must satisfy my specific `MRT_RowData` type for the table, so any props marked as required must be in the default data. 

## Solution
Can we wrap the type of this param in a `Partial` to let you provide default values for only a subset of columns? I'm getting no errors in the actual running instance of the component despite the TS warning so have had to add a `// @ts-expect-error` in the mean time.

**Comment by KevinVandy** [comment]
Sure. PR?

**Reply by sdriffill-tm** [reply]
Will do 😄
